[DRAFT] feat(cx-navigation): Add Cx Navigation

gzeni-cit opened the pull request:
<!--
  Before issuing a pull request:

  - Please read our contribution guidelines (https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/packages/bodiless-documentation/doc/Development/Contributing.md), especially the section on Pull Requests.
  - Please first create an issue (https://github.com/johnsonandjohnson/Bodiless-JS/issues/new). All pull requests must be linked to an issue.
-->

<!--
  IMPORTANT
  
  The pull request title will become the commit message title at merge, and
  should adhere to Angular Commit Message Conventions. Please see (https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/packages/bodiless-documentation/doc/Development/Contributing.md) for details and examples.
-->

## Changes
<!-- Brief description of the changes introduced by this PR -->

## Test Instructions
<!-- Detailed description of how this feature was (and should be) tested
  - Setup instructions (if any)?
  - List of test cases, with steps and expected results?
  - List
-->

## Related Issues
<!--
  Link to the issue that is fixed or resolved by this PR (if there is one)
  e.g. Fixes #1234, Closes #4567

  Link to an issue that is partially addressed by this PR (if there are any)
-->

<!-- IF THIS PR INTRODUCES A BREAKING CHANGE

BREAKING CHANGE: Describe the nature of the breaking change here.

More Details about the breaking change.
-->

Commit:
feature(cx-navigation): Add CX Navigation

hvanyo commented on packages/cx-navigation/src/components/BurgerMenu/tokens/cxBurgerMenu.ts:
```diff
@@ -0,0 +1,91 @@
+  Behavior: {
+    Wrapper: flowHoc(
+      addClassesIf(useIsBurgerMenuHidden)('transform -translate-x-full'),
+      // @TODO: Maybe, put custom ease in tailwind theme configs?
```
@Guilherme-Almeida-Zeni  I actually prefer it here as dev its quicker to adjust and not have to restart site.    So I think I wouldn't move to tailwind.   Alternative we could make it token if we think we might push something else but I can't think of any other component that will do this so its one off and doesn't make sense as token either.

hvanyo commented on packages/cx-navigation/src/components/BurgerMenu/tokens/cxBurgerMenu.ts:
```diff
@@ -0,0 +1,91 @@
+      replaceWith(Span),
+      withProps({
+        children: 'Español',
+        className: 'px-4 border-l-2 border-gray-400',
```
@Guilherme-Almeida-Zeni  I think all colors need to be a color token so we can shadow them.  Is this not the cxSeparator vertical line used elsewhere?  and reuse that token?

hvanyo commented on packages/cx-navigation/src/components/Menu/tokens/cxMenu.ts:
```diff
@@ -14,12 +31,15 @@ import { cxSeparator } from '../../Separator';
  */
 const Base = asMenuToken({
-  // @todo do we need this?
+  // @TODO: Do we need this?
```
@Guilherme-Almeida-Zeni  yes -- its simple token that converts menu title text to attribute -> https://sourcecode.jnj.com/projects/ASX-NBFP/repos/canvasx/browse/packages/canvasx-navigation/src/util/withMenuAnalytics.tsx#11,15    (This is the GA4 requirement in ACs.)

hvanyo commented on packages/cx-navigation/src/components/Separator/tokens/cxSeparator.ts:
```diff
@@ -40,6 +40,19 @@ const FooterMenu = asElementToken({
+    },
+  },
+  Theme: {
+    _: 'px-4 border-l-2 border-gray-400',
```
@Guilherme-Almeida-Zeni  another separator with color -- needs to be token easily shadowed.  As well as color as separate token.

Commit:
feature(cx-navigation): Add WTB Button Placeholder

Commit:
feature(navigation): Update Content For Better Styling

Commit:
feature(bc-navigation): Style Burger Menu For Mobile And Tablet

wodenx commented on packages/cx-layout/src/components/WhereToBuy/WhereToBuyClean.tsx:
```diff
@@ -0,0 +1,47 @@
+const socialLinksComponents: WhereToBuyComponents = {
+  Wrapper: Div,
+  Button: A,
+  Icon: CartIcon,
```
I wonder if instead of starting with this we should add it in the content section of a default token

Commit:
feature(bc-navigation): Fix Where To Buy Components Name

Commit:
Merge branch 'feature/bc-footer' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

Commit:
feature(bc-navigation): Fix Issues Due To Conflict Resolution

Commit:
feature(bc-navigation): Reupdate Header After Conflicts Resolution

Commit:
feature(bc-navigation): Add Navigation Desktop Style

Commit:
Merge branch 'feature/bc-footer' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

Commit:
feature(navigation): Remove Unused Module Import

Commit:
feature(bc-navigation): Add NoWrap To Menus

Commit:
feature(bc-navigation): Address Feedback

Commit:
Merge branch 'feature/bc-footer' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

wodenx committed:
Feature/bc navigation (#19)

* Replace WTB with styled link

* Fix lint

wodenx committed:
Feature/cx navigation toggler (#20)

* Replace WTB with styled link

* Fix lint

* Remove burger menu toggler component.

Co-authored-by: Guilherme Almeida Zeni <gzeni@ciandt.com>

Commit:
feature(bc-navigation): Last Updates

gzeni-cit committed:
feat(bc-footer): Add Brand.com Footer (#1450)

* feature(bc-footer): Add Brand.com Footer

* feature(bc-footer): Final Styles Adjustments

* feature(bc-footer): Copyright as RTE, Add TODO On Social Links AND Removed Unnecessary FlowHoc

* feature(bc-footer): Move Sepator Styles To Its Own Token

* feature(bc-footer): Rename GlobalFooter To Footer

* feature(bc-footer): Address PR Feedback - Batch 1

* Reorganize menu. (#7)

* Reorganize menu.

* Remove redundant footer menu styling.

* Add submenu titles.

* Remove duplication in MenuTitleClean

* Fix lint

* Add canvasx token docs.

* feature(bc-footer): Fix doc typo

Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

* feature(bc-footer): Fix doc typo

Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

Co-authored-by: Guilherme Almeida Zeni <gzeni@ciandt.com>
Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

* feature(bc-footer): Update Footer Menu Data

* feature(bc-footer): Update Package.Lock

* feature(bc-footer): Remove Console Log From Menu Separator

* feature(bc-footer): Improve Footer Structure For Later Static Implementation

* feature(bc-footer): Add Todo Comments

* feature(bc-footer): Rename SiteHeader And SiteFooter To Header And Footer

* feature(bc-footer): Add Cx Navigation To TSConfig

* feature(bc-footer): Rename Remaining SiteHeader And SiteFooter To Header And Footer

* feature(bc-footer): Adjust Layout After Updates

* Token site constrains (#8)

* feature(bc-footer): Add Footer Slots Wrappers

* feature(bc-footer): Refactor Footer Copyright RichText

* feature(bc-footer): Update Footer And Copyright Keys

* feature(bc-footer): Add Footer Page Example

* feature(bc-footer): Feedback Fixes - Batch 2

* feature(bc-footer): Update Cx Navigation Package Version

* fix(bc-footer): Fix Footer Issues (#10)

* bugfix(bc-footer): Fix Footer Issues

* Remove top margin from in footnote

* Add uppercase to submenu items

* Removing unused imports

* bugfix(bc-footer): Fix comments on PR

* bugfix(bc-footer): Fix link issues

Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>
Co-authored-by: guilhermegpessoa <guilherme.gpessoa@hotmail.com>

* feature(bc-footer): Fix issues after main branch update

* fix(bc-footer): Fix footer issues (#13)

* fix(bc-footer): Fix padding issues

* fix(bc-footer) Fix padding values

* fix(bc-footer): Fix padding issues

* feature(bc-footer): Update Footer Content

* bugfix(bc-footer): Fix Footer Menu Issues Issues (#16)

* bugfix(bc-footer): Fix Footer Menu Accessibility Issues

* bugfix(bc-footer): Add External And Download Icons To Menu Title Links

Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>

* bugfix(bc-footer): Fix Mobile Margins Between Submenus (#17)

Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>

Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>
Co-authored-by: wodenx <wodenx@gmail.com>
Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>
Co-authored-by: guilhermegpessoa <guilherme.gpessoa@hotmail.com>
Co-authored-by: guilhermegpessoa <48322825+guilhermegpessoa@users.noreply.github.com>

Commit:
Merge branch 'main' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

Commit:
feature(bc-navigation): Remove Old Index File

misterjones committed:
docs(cx-elements, cx-layout): Update Token Shadowing docs (#1484)

* Update Token Shadowing documentation

- Split Token Shadowing out of "CX Component Template" page into
  separate page.
- Perform standard doc cleanup (typos, appease the linter, 100 col
  width, etc.).

* Update shadow paths

* Restructure pages

gzeni-cit committed:
Merge branch 'main' into feature/bc-navigation

lucas-varela committed:
test(hydration): Add withoutHydration unit tests (#1488)

* Add unit tests to withoutHydration

* Remove extra line on server side test

* Move react-dom to peerDependencies

gzeni-cit committed:
Merge branch 'main' into feature/bc-navigation

Commit:
feature(bc-navigation): Update Header Constrains To Be In Container Component

Commit:
Merge branch 'feature/bc-navigation' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

Asemirski committed:
feat: action to crate daily qa branch implemented (#1494)

gzeni-cit committed:
Merge branch 'main' into feature/bc-navigation

dewen committed:
fix(gatsby-theme-bodiless): remove null-loader package dep. (#1493)

Co-authored-by: Dewen Li <dli94@its.jnj.com>
Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

gzeni-cit committed:
Merge branch 'main' into feature/bc-navigation

wodenx committed:
fix(gatsby-theme-bodiless): Properly prefix paths for bodiless gatsby… (#1502)

## Changes
Applies path prefix (if active) to all gatsby image urls

## Related Issues
Fixes #1501

wodenx committed:
feat(cx-templates, cx-flowcontainer): Add generic template  (#1483)

* WIP on generic template

* Fix issue with nested tokens (#1455)

* Clean template & start shadow

* Cleanup and renaming

* Some cleanup, remove hero flow container.

* Add gutter to content region

* add copyright

* Fix broken styleguide tokens.

* fix some issues after merge

* Fix lint

* move wrapper

* 8512: Add design keys for testing identifiers

* 8541 Add Breadcrumb placeholder & style

* AESQ-8536 add border to demonstrate contentregion

* 8543 Fix lint & add tailwind

* Correct name of footer styleguide page.

* 8537

* 8549 & 8550

* fix build error

* Add tailwind to cx-flowcontainer

* remove dup of generic template & header/footer

* fix build error

* Single Constrained Flow container & fragments

* Move SiteContraint from Wrapper -> container

* Move breadcrumb over hero/top

* comment out cypress

Co-authored-by: hvanyo <hvanyo@its.jnj.com>
Co-authored-by: tmiraniuk <tmiraniu@its.jnj.com>

Commit:
Merge branch 'main' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

Commit:
feature(bc-navigation): Fix Header BG After Main Updates

lucas-varela committed:
chore(release): publish v1.0.0-beta.2

lucas-varela committed:
Update package locks

Commit:
docs(bc-navigation): BC Navigation Docs

lucas-varela committed:
Merge pull request #1517 from johnsonandjohnson/release

chore(release): publish v1.0.0-beta.2

gzeni-cit committed:
bugfix(bc-navigation): Fix Navigation Issues (#21)

Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>

gzeni-cit committed:
Merge branch 'main' into feature/bc-navigation

Commit:
feature(bc-navigation): Fix Submenu Paddings

misterjones committed:
docs(cx-elements): Brand.com Footer (#1506)

* Update Token documentation

* Update Token documentation included in the Brand.com Footer work.
  * Change instances of "CanvasX" to "Vital."
  * Move doc files to top-level of `doc` directory.
  * Update JSON file to include Token docs.

* Update docs based on feedback

* Update packages/cx-layout/doc/Footer.md

* Update packages/cx-layout/doc/Footer.md

Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

gzeni-cit committed:
bugfix(bc-navigation): Fix Safari Burger Menu Issue And Add ID To Header (#22)

Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>

gzeni-cit committed:
Merge branch 'main' into feature/bc-navigation

gzeni-cit committed:
feature(bc-navigation): Fix Safari Issues (#24)

Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>

konst8 committed:
chore!(fclasses, cli): Standard Module Resolution

## Changes
Change autodiscovery approach for:
- tailwind configs
- docs
- environment variables configs

## Related Issues
Closes #1277

Asemirski committed:
ci: prevent PR merge if commit message is not semantic (#1537)

Action that performs sematic check implemented.

Co-authored-by: Aliaksei Semirski <aliaksei_semirski@epam.com>

wodenx committed:
WIP on i18n

Commit:
Merge branch 'main' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

Commit:
feature(cx-navigation): Update Tailwind Configs After Merge Conflicts Resolution

wodenx committed:
Revert "WIP on i18n"

This reverts commit 316abe52c7fde72eefce07411954829d3bbdbbf8.

lucas-varela committed:
fix(fclasses): Fix replaceWith passing extra props to Fragments (#1530)

wodenx committed:
fix: Correct exports of package tailwind configs and starter activation. (#1538)

* fix: Correct exports of package tailwind configs and starter activation.

* Ensure tailwind.config.js is exported.

* Add 'getDocs.js' to files section of package.json

* fix: update cx-layout tailwind-config

Co-authored-by: Konstantin Mishur <kmishur@its.jnj.com>
Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

Commit:
Merge branch 'main' of github.com:Guilherme-Almeida-Zeni/Bodiless-JS into feature/bc-navigation

lucas-varela committed:
chore(release): publish v1.0.0-beta.3

Commit:
feature(bc-navigation): Fix Top Nav Menu Style

lucas-varela committed:
Update package files

lucas-varela committed:
Merge pull request #1545 from johnsonandjohnson/release

chore(release): publish v1.0.0-beta.3

gzeni-cit committed:
Merge branch 'main' into feature/bc-navigation
